query/sources/impl/src/main/java/org/codice/ddf/admin/sources/commons/SourceActionCommons.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.commons;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.admin.common.fields.base.ListFieldImpl;
+import org.codice.ddf.admin.common.fields.common.PidField;
+import org.codice.ddf.admin.configurator.ConfigReader;
+import org.codice.ddf.admin.configurator.ConfiguratorFactory;
+import org.codice.ddf.admin.sources.fields.SourceInfoField;
+import org.codice.ddf.admin.sources.fields.type.SourceConfigUnionField;
+
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.ConnectedSource;
+import ddf.catalog.source.FederatedSource;
+import ddf.catalog.source.Source;
+
+public class SourceActionCommons {
+
+    public static SourceInfoField createSourceInfoField(boolean isAvailable, SourceConfigUnionField config) {
+        config.credentials().password("*****");
+        SourceInfoField sourceInfoField = new SourceInfoField();
+        sourceInfoField.isAvaliable(isAvailable);
+        sourceInfoField.configuration(config);
+        return sourceInfoField;
+    }
+
+    /**
+     * Gets the configurations for the given factoryPids using the {@link ConfiguratorFactory}. A mapper is used
+     * to transform the service properties to a {@link SourceConfigUnionField}.
+     *
+     * @param factoryPids factory pids to lookup configurations for
+     * @param mapper a {@link Function} taking a map of string to objects and returning a {@code SourceConfigUnionField}
+     * @param selector a servicePid to select a single configuration, returns all configs when null or empty
+     * @return a list of {@code SourceInfoField}s configured in the system
+     */
+    public static ListFieldImpl<SourceInfoField> getSourceConfigurations(List<String> factoryPids, Function<Map<String, Object>, SourceConfigUnionField> mapper,
+            PidField selector, ConfiguratorFactory configuratorFactory) {
```

Comment:
Is there a reason we are using Field for the pid? We could pass a string instead.